### PR TITLE
Add utils functions

### PR DESCRIPTION
**Pull request by pabloarosado**
I just created a "utils" module (although the name may be redundant). In this module I have put some of the functions I created for the energy-data repos, that I find useful, and quite generic.

### CODE REVIEW

**Comment by lucasrodes** [comment]
Thanks for your contribution, @pabloarosado! I think overall this can be a _very_ useful module!

Some notes:
- It was not passing the CI build, mainly due to missing typings. I've added those so I could run the tests and the CI/CD eventually can pass. You can always check this by running `make test` before pushing.
- I think we should probably have another name for the module. As it is very specific to `pd.DataFrames` we could have it renamed to `dataframes.py`? I feel like we will have other utils for files, web, visualizations etc., so probably good to have them in separate modules.

**Comment by pabloarosado** [comment]
Hi @lucasrodes, thanks, great suggestions. I've created a `common` module (I thought `datautils.utils` was a bit redundant) for common, generic objects. Then I have also created the `dataframes` module, as you suggested. Then I realised that the functions had unnecessarily long names, so I simplified them.

**Comment by lucasrodes** [comment]
Hi @pabloarosado, great, thanks for the additions! Just some thoughts on the naming proposals:

### Module name
* I would reserve `datautils.common` (or similar naming) for library internal tools, or code that appears often in many library modules.
* For code related to tools that we often use I would have a dedicated module, as we currently do for `s3`. Otherwise, I find it slightly redundant to have a 'common' module within a 'common' library for 'common' tools. For instance, in such case, it is not clear to me why `s3` code would be not within `datautils.common`. 
* Instead, I would use `datautils.dataframes` or `datautils.df`. It could either come as a single file (i.e. `dataframe.py`) or as a whole submodule (i.e. `datautils/dataframe/__init__.py`, `datautils/dataframe/utils.py`).

### Renaming of functions
* While I think having not-to-long names is great, I think that they should be self-explanatory.
* From the context and the argument types, it is easy to see what `compare` does, but:
  * I imagine a case where we may have two separate modules with functions with the same name. Suppose we have a module that compares `pandas.Series` objects. If you needed both functions in your code, when importing them and to avoid collisions you'd have to:

  ```python
  from datautils.series import compare as compare_series
  from datautils.dataframes import compare as compare_dataframes
  ```

  Which slightly adds some overhead.

File: owid/datautils/dataframes.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+class ObjectsAreNotDataframes(ExceptionFromDocstring):
+    """Given objects are not dataframes."""
+
+
+def compare(
+    df1: pd.DataFrame,
+    df2: pd.DataFrame,
+    columns: Union[List[str], None] = None,
+    absolute_tolerance: Union[int, float] = 1e-8,
```
Comment (Marigold) [inline owid/datautils/dataframes.py:24]: Using `absolute_tolerance: float` should be sufficient.

File: owid/datautils/common.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""Common objects shared by other modules.
+
+"""
+
+from typing import Union, Any
+
+
+class ExceptionFromDocstring(Exception):
```
Comment (Marigold) [inline owid/datautils/common.py:8]: Neat trick!

File: owid/datautils/dataframes.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+
+class ObjectsAreNotDataframes(ExceptionFromDocstring):
+    """Given objects are not dataframes."""
+
+
+def compare(
+    df1: pd.DataFrame,
+    df2: pd.DataFrame,
+    columns: Union[List[str], None] = None,
```
Comment (Marigold) [inline owid/datautils/dataframes.py:23]: This could be written as `Optional[List[str]]`

**Review by Marigold — APPROVED** [review]

**Comment by Marigold** [comment]
@lucasrodes when working with utils modules I personally import them with an alias rather than importing individual functions. So for me it wouldn't really matter. Both approaches have their pros and cons, it's hard to tell what is better.

```python
from datautils import series as utils_series
from datautils import dataframes as utils_df

utils_df.compare(...)
```

**Review by lucasrodes — APPROVED** [review]
Nice, Thanks for sharing your thoughts & implementing this @pabloarosado @Marigold

File: owid/datautils/utils.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+        abs(a - b) <= absolute_tolerance
+    relative_tolerance : float
+        Relative tolerance to assume in the comparison of each cell in the dataframes. A value a of an element in df1 is
+        considered equal to the corresponding element b at the same position in df2, if:
+        abs(a - b) / abs(b) <= relative_tolerance
+
+    Returns
+    -------
+    compared : pd.DataFrame
+        Dataframe with as many rows as df1 and df2, and as many columns as specified by `columns` argument (or as many
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply owid/datautils/utils.py]: Thanks. Feel free to make this changes yourself and merge.

**Pull request merged**